x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -99,63 +120,266 @@ export class ActionExecutor {
+        });
+      },
+      executeLabel: `execute_action`,
+      executionId,
+      isEphemeral,
+      namespace,
+      params,
+      relatedSavedObjects,
+      request,
+      services,
+      source,
+      spaceId,
+      taskInfo,
+    });
+  }
+
+  public async executeUnsecured({
+    actionExecutionId,
+    actionId,
+    params,
+    relatedSavedObjects,
+    spaceId,
+    source,
+  }: UnsecuredExecuteOptions): Promise<ConnectorTypeExecutorResult<unknown>> {
+    const { getUnsecuredServices } = this.actionExecutorContext!;
+
+    const services = getUnsecuredServices();
+    const namespace = spaceId && spaceId !== 'default' ? { namespace: spaceId } : {};
+
+    return await this.executeHelper({
+      actionExecutionId,
+      actionId,
+      executeLabel: `execute_unsecured_action`,
+      namespace,
+      params,
+      relatedSavedObjects,
+      services,
+      source,
+      spaceId,
+    });
+  }
+
+  public async logCancellation<Source = unknown>({
+    actionId,
+    request,
+    relatedSavedObjects,
+    source,
     executionId,
+    taskInfo,
     consumer,
-    relatedSavedObjects,
     actionExecutionId,
-  }: ExecuteOptions): Promise<ActionTypeExecutorResult<unknown>> {
+  }: {
+    actionId: string;
+    actionExecutionId: string;
+    request: KibanaRequest;
+    taskInfo?: TaskInfo;
+    executionId?: string;
+    relatedSavedObjects: RelatedSavedObjects;
+    source?: ActionExecutionSource<Source>;
+    consumer?: string;
+  }) {
+    const { spaces, eventLogger } = this.actionExecutorContext!;
+
+    const spaceId = spaces && spaces.getSpaceId(request);
+    const namespace = spaceId && spaceId !== 'default' ? { namespace: spaceId } : {};
+
+    if (!this.connectorData || this.connectorData.connectorId !== actionId) {
+      this.connectorData = await this.getConnectorData(actionId, namespace.namespace);
+    }
+
+    const task = taskInfo
+      ? {
+          task: {
+            scheduled: taskInfo.scheduled.toISOString(),
+            scheduleDelay: Millis2Nanos * (Date.now() - taskInfo.scheduled.getTime()),
+          },
+        }
+      : {};
+    // Write event log entry
+    const event = createActionEventLogRecordObject({
+      actionId,
+      consumer,
+      action: EVENT_LOG_ACTIONS.executeTimeout,
+      message: `action: ${this.connectorData.actionTypeId}:${actionId}: '${
+        this.connectorData.name ?? ''
+      }' execution cancelled due to timeout - exceeded default timeout of "5m"`,
+      ...namespace,
+      ...task,
+      executionId,
+      spaceId,
+      savedObjects: [
+        {
+          type: 'action',
+          id: actionId,
+          typeId: this.connectorData.actionTypeId,
+          relation: SAVED_OBJECT_REL_PRIMARY,
+        },
+      ],
+      relatedSavedObjects,
+      actionExecutionId,
+      isInMemory: this.connectorData.isInMemory,
+      ...(source ? { source } : {}),
+    });
+
+    eventLogger.logEvent(event);
+  }
+
+  public async getConnectorData(
```

Comment:
renamed `getActionInfoInternal` to `getConnectorData` since this function is public and used outside of this class.